upcgen.sh

```diff
@@ -24,6 +24,7 @@ mkdir -p $INSTALLROOT/bin
 cp ./upcgen $INSTALLROOT/bin/
 mkdir -p $INSTALLROOT/lib
 cp ./libUpcgenlib.a $INSTALLROOT/lib/.
+cp ./libUpcgenlib.so $INSTALLROOT/lib/.
```

Comment:
Indeed, you should add double quotes around the various `$INSTALLROOT/something`.